tools/src/edm4hep2json.cxx

```diff
@@ -120,20 +115,7 @@ int main(int argc, char** argv) {
     outFilePath = std::filesystem::path(outFileStr + ".edm4hep.json");
   }
 
-  bool legacyReader = false;
-  {
-    std::unique_ptr<TFile> inFile(TFile::Open(inFilePath.c_str(), "READ"));
-    legacyReader = !inFile->GetListOfKeys()->FindObject("podio_metadata");
-  }
-
-  if (legacyReader) {
-    std::cout << "WARNING: Reading legacy file, some collections might not be recognized!" << std::endl;
```

Comment:
Can we keep the check and a slightly adapted warning message that tells people that they cannot use this file because it has not been done with a compatible EDM4hep version?

Reply:
Check added back and modified, now it's an error since it's likely going to crash anyway.